### PR TITLE
Xtensor template fix

### PR DESCRIPTION
**Pull request by jtramm**
This is a workaround for newer versions of LLVM that are having some sort of interaction with xtensor that is causing issues. Notably, this will also affect the main development branch of OpenMC eventually, though is much more of a present issue for the OMP fork given our reliance on newer compilers.

The intertwined issues are a little hard to parse, but I'm gathering that the issue may be in xtensor itself rather than with LLVM, but I'm not 100% on that.

Thankfully, there is an easy workaround flag that can be passed to LLVM that fixes the error. I think we should add it in now so that our auto testing works again and I'll plan on checking manually now and then to see if the flag is still required or not (and/or if an update to the xtensor submodule fixes it).

See issues:
- https://github.com/llvm/llvm-project/issues/91504
- https://github.com/xtensor-stack/xtensor/issues/2783

### CODE REVIEW

**Review by paulromano — APPROVED** [review]
Thanks!

**Comment by gridley** [comment]
Note: I was running into this tonight building openmc on my mac. the LLVM homebrew gives you now runs into this.

Note on the note: this flag is deprecated but it fixes the issue nonetheless.

clang++: warning: argument '-fno-relaxed-template-template-args' is deprecated [-Wdeprecated-no-relaxed-template-template-args]

**Comment by emmenlau** [comment]
See https://github.com/xtensor-stack/xtensor/pull/2821 for work towards a potential fix.
